catalog/controller/extension/payment/razorpay.php

```diff
@@ -0,0 +1,132 @@
+
+            $order_info = $this->model_checkout_order->getOrder($merchant_order_id);
+            $amount = $this->currency->format($order_info['total'], $order_info['currency_code'], $order_info['currency_value'], false) * 100;
+
+            $success = false;
+            $error = '';
+
+            try {
+                $ch = $this->get_curl_handle($razorpay_payment_id, $amount);
+
+                    //execute post
+                    $result = curl_exec($ch);
+                $http_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+                if ($result === false) {
+                    $success = false;
+                    $error = 'Curl error: '.curl_error($ch);
+                } else {
+                    $response_array = json_decode($result, true);
+                        //Check success response
+                        if ($http_status === 200 and isset($response_array['error']) === false) {
+                            $success = true;
+                        } else {
+                            $success = false;
+
+                            if (!empty($response_array['error']['code'])) {
+                                $error = $response_array['error']['code'].':'.$response_array['error']['description'];
+                            } else {
+                                $error = 'RAZORPAY_ERROR:Invalid Response <br/>'.$result;
+                            }
+                        }
+                }
+
+                    //close connection
+                    curl_close($ch);
+            } catch (Exception $e) {
+                $success = false;
+                $error = 'OPENCART_ERROR:Request to Razorpay Failed';
+            }
+
+            if ($success === true) {
+                if (!$order_info['order_status_id']) {
+                    $this->model_checkout_order->addOrderHistory($merchant_order_id, $this->config->get('razorpay_order_status_id'), 'Payment Successful. Razorpay Payment Id:'.$razorpay_payment_id);
+                } else {
+                    $this->model_checkout_order->addOrderHistory($merchant_order_id, $this->config->get('razorpay_order_status_id'), 'Payment Successful. Razorpay Payment Id:'.$razorpay_payment_id);
+                }
+
+                echo '<html>'."\n";
+                echo '<head>'."\n";
+                echo '  <meta http-equiv="Refresh" content="0; url='.$this->url->link('checkout/success').'">'."\n";
+                echo '</head>'."\n";
+                echo '<body>'."\n";
+                echo '  <p>Please follow <a href="'.$this->url->link('checkout/success').'">link</a>!</p>'."\n";
+                echo '</body>'."\n";
+                echo '</html>'."\n";
+                exit();
+            } else {
+                $this->model_checkout_order->addOrderHistory($this->request->request['merchant_order_id'], 10, $error.' Payment Failed! Check Razorpay dashboard for details of Payment Id:'.$razorpay_payment_id);
+                echo '<html>'."\n";
```

Comment:
If we are echoing the first byte of html, why can't we send a location header instead?